playbooks/resize_disk.yaml

```diff
@@ -0,0 +1,31 @@
+---
+- hosts: all
+  become: true
+  tasks:
+    - name: Install cloud-utils-growpart package
+      package:
+        name: cloud-utils-growpart
+
+    - name: Install e2fsprogs package
+      package:
+        name: e2fsprogs
+      when: ansible_distribution_major_version == '7'
+
+    - name: Extend partition
+      shell: growpart /dev/vda 1
+      register: output
+      ignore_errors: true
+
+    - name: Fail if growpart indicates nochange
+      fail:
+      when:
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc is not 0
+
+    - name: Grow XFS filesystem
+      shell: xfs_growfs -d /
+      when: ansible_distribution_major_version == '8'
+
+    - name: Grow ext4 filesystem
+      shell: resize2fs /dev/vda1
+      when: ansible_distribution_major_version == '7'
```

Comment:
Isn't it better to detect the partition type and resize it accordingly?

Reply:
I don't know. I could try that.